File: frontend/src/types/knowledge-base.ts
export interface KBDocument {
  id: number;
  user_id: number;
  slug: string;
  title: string;
  content: string;
  is_default: boolean;
  used_by: string[];
  created_at: string;
  updated_at: string;
}

export interface KBDocumentCreateInput {
  slug: string;
  title: string;
  content: string;
  used_by: string[];
}

export interface KBDocumentUpdateInput {
  title?: string;
  content?: string;
  used_by?: string[];
}

File: backend/app/models/knowledge_base.py
from datetime import datetime

from sqlalchemy import Boolean, DateTime, ForeignKey, Integer, JSON, String, Text, UniqueConstraint, func
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class AiKnowledgeBase(Base):
    __tablename__ = "ai_knowledge_base"

    id: Mapped[int] = mapped_column(primary_key=True)
    user_id: Mapped[int] = mapped_column(
        Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True
    )

    slug: Mapped[str] = mapped_column(String(100), nullable=False)
    title: Mapped[str] = mapped_column(String(255), nullable=False)
    content: Mapped[str] = mapped_column(Text, nullable=False)
    is_default: Mapped[bool] = mapped_column(Boolean, nullable=False, server_default="true")
    used_by: Mapped[list] = mapped_column(JSON, default=list, nullable=False, server_default="[]")

    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False
    )

    __table_args__ = (
        UniqueConstraint("user_id", "slug", name="uq_kb_user_slug"),
    )
